### ISSUE TITLE
NVDA should skip disabled add-ons when creating portable or system copies

### ISSUE THREAD

**Issue by ultrasound1372** [nvaccess/nvda#9020]
There is another issue similar to this one open, but it only covers system config and it seems to have not been looked at. Since the disabled add-ons facility was recently added, I figured why not bring this up? I'd like to see this in 2018.4.
I, and some other users probably, have some large add-ons such as vocalizer or acapella voices, or other synth drivers and such I do not need on portable copies for school computers. In order to skip copying those now and taking forever to do so, I must cut the add-ons into a temporary storage folder and restart NVDA, create the copy, put them back and restart again. Since we have a facility to disable add-ons I thought why not simply skip copying them if they are disabled? Still requires a restart but you don't have to manually move things around. The same should apply to system config when copying add-ons, I don't need most of my add-ons in system config but I might need a few, like some people need remote, or my primary synth driver, etc.

**Comment by josephsl** [comment]
Hi, sorry, it won’t make the cut for 2018.4. I’ll take this on but after #6275 is merged into master first. CC @NVDAEs in case she wants to take this on, as checkable list is a good implementation strategy for this. Thanks.

**Comment by Brian1Gaff** [comment]
Maybe the skip or not should be a setting when you are asked questions when 
you are about to create a portable version. In some cases you may want them 
after all.
 Also has any work been done that might be able to mitigate the issues of 
reduced functionality on temp copies in windows 10?
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by josephsl** [comment]
Hi,

Ping @NVDAEs as all the necessary things are present.

Thanks.

**Comment by ultrasound1372** [comment]
A checkable list would be a good idea for this yes. When you create a system copy and it warns about custom plugins, you can hit copy all, don't copy, or select which to copy. And when creating portable copies, when you check the box a button will appear, choose add-ons... That brings up a dialog with a checkable list where all enabled add-ons are checked by default and disabled ones are unchecked. Then you can modify that as needed, hit OK, other button is cancel, then hit create.

**Comment by Adriani90** [comment]
cc: @josephsl, @nvdaes are you still thinking to implement this, now that everything is in place from the technical part?
